pkg/v1/remote/check.go

```diff
@@ -18,13 +19,13 @@ import (
 func CheckPushPermission(ref name.Reference, kc authn.Keychain, t http.RoundTripper) error {
 	auth, err := kc.Resolve(ref.Context().Registry)
 	if err != nil {
-		return err
+		return fmt.Errorf("Resolving authorization for %v failed: %v", ref.Context().Registry, err)
```

Comment:
Small nit, can you make these error strings start with lower case? See https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
Done.